core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -143,9 +180,9 @@ public int asDistance() {
     public int asDistance(String text) {
         if (this.equals(AUTO)) { //AUTO
             final int len = termLen(text);
-            if (len <= 2) {
+            if (len < lowDistance) {
                 return 0;
-            } else if (len > 5) {
+            } else if (len >= highDistance) {
```

Comment:
Also I think we'd need to update the logic so that it is easier to explain and more future-proof in case we ever want to add a 3rd distance for 3 edits. So maybe:
 - make the logic look like this:
```
if (len < lowDistance) {
  return 0;
} else if (len < highDistance) {
  return 1;
} else {
  return 2;
}
```
 - change `DEFAULT_LOW_DISTANCE` to 3 and `DEFAULT_HIGH_DISTANCE` to 6 so that defaults remain the same
 - add docs for it to the `fuzziness` section of `docs/reference/api-conventions.asciidoc`